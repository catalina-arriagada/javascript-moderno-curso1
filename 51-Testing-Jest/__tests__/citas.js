import Citas from '../js/classes/Citas';

describe('probar clase Citas', () => {
    
    const citas = new Citas();
    const id = Date.now();

    test('test para agregar nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'j',
            telefono: '111',
            fecha: '10-12-10',
            hora:'10:30',
            sintomas: 'aaa'
        };
        citas.agregarCita(citaObj);

        //Escribir prueba
        expect(citas).toMatchSnapshot(); //que clase se esta guardando correctamente
        
    })
    test('actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'nuevo nombre',
            propietario: 'j',
            telefono: '111',
            fecha: '10-12-10',
            hora:'10:30',
            sintomas: 'aaa2'
        };
        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    })
    test('eliminar cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    })
})

//npm t -- -u