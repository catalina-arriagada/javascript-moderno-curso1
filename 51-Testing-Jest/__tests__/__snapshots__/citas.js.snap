// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar clase Citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-10",
      "hora": "10:30",
      "id": 1706668598307,
      "mascota": "nuevo nombre",
      "propietario": "j",
      "sintomas": "aaa2",
      "telefono": "111",
    },
  ],
}
`;

exports[`probar clase Citas eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`probar clase Citas test para agregar nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-10",
      "hora": "10:30",
      "id": 1706668598307,
      "mascota": "Hook",
      "propietario": "j",
      "sintomas": "aaa",
      "telefono": "111",
    },
  ],
}
`;
